presto/presto.go

```diff
@@ -1205,19 +1201,18 @@ func unmarshalArguments(signature *typeSignature) error {
 			payload = &(signature.Arguments[i].namedTypeSignature)
 		case KIND_LONG:
 			payload = &(signature.Arguments[i].long)
+		default:
+			return fmt.Errorf("unknown argument kind: %s", argument.Kind)
 		}
-		err := json.Unmarshal(argument.Value, payload)
-		if err != nil {
```

Comment:
Without this new `default` case, the `json.Unmarshal` call would return a confusing error, because we would attempt to unmarshal into a `nil` pointer, which isn't valid. The new `default` case` at least gives us a clearer error.